yt_dlp/extractor/owncloud.py

```diff
@@ -0,0 +1,88 @@
+import urllib.parse
+
+from yt_dlp.extractor.common import InfoExtractor
+from yt_dlp.utils import (
+    determine_ext,
+    ExtractorError,
+    url_or_none,
+    urlencode_postdata,
+)
+
+
+class OwnCloudIE(InfoExtractor):
+    _INSTANCES_RE = r'''(?:
+                            sciebo\.de|
+                            [^\.]+\.sciebo\.de|
+                            cloud\.uni-koblenz-landau\.de|
+                        )'''
+    _VALID_URL = rf'''(?x)
+        (?P<server>https?://{_INSTANCES_RE})/s/
+        (?P<id>[\w\-\.]+)'''
```

Comment:
```suggestion
    _INSTANCES_RE = '|'.join((
        r'(?:[^\.]+\.)?sciebo\.de',
        r'cloud\.uni-koblenz-landau\.de',
    ))
    _VALID_URL = f'(?P<url>https?://(?:{_INSTANCES_RE})/s/(?P<id>[\w.-]+))'
```
See below comments for use of `url`